Forms/Fields/DateOfBirthDropdownField.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DUCK.Forms.Fields
+{
+	public class DateOfBirthDropdownField : AbstractFormField
+	{
+		[SerializeField]
+		private Dropdown dayDropdown;
+		[SerializeField]
+		private Dropdown monthDropdown;
+		[SerializeField]
+		private Dropdown yearDropdown;
+
+		[SerializeField]
+		private Color onValueChangedColor;
+
+		private int defaultDay;
+		private int defaultMonth;
+		private int defaultYear;
+
+		protected override void Awake()
+		{
+			base.Awake();
+
+			defaultDay = dayDropdown.value;
+			defaultMonth = monthDropdown.value;
+			defaultYear = yearDropdown.value;
+
+			dayDropdown.onValueChanged.AddListener(value => { dayDropdown.captionText.color = onValueChangedColor; });
+			monthDropdown.onValueChanged.AddListener(value =>
+			{
+				monthDropdown.captionText.color = onValueChangedColor;
+				HandleValueChanged(value);
+			});
+			yearDropdown.onValueChanged.AddListener(value =>
+			{
+				yearDropdown.captionText.color = onValueChangedColor;
+				HandleValueChanged(value);
+			});
+
+			monthDropdown.AddOptions(DateOfBirthUtils.Months.ToList());
+			dayDropdown.AddOptions(DateOfBirthUtils.Days.Select(day => day.ToString()).ToList());
+			yearDropdown.AddOptions(DateOfBirthUtils.Years.Select(year => year.ToString()).ToList());
+
+			HandleValueChanged(monthDropdown.value);
+		}
+
+		public override object GetValue()
+		{
+			return new DateOfBirth(GetDay(), GetMonth(), GetYear());
+		}
+
+		private void HandleValueChanged(int value)
```

Comment:
You might as well not pass this value in, it's not used

Reply:
It needs to match the delegates signature for the Dropdown API

Reply:
I would rename this function `UpdateDaysInMonth` or something similar

Reply:
Nevermind, it did need to, ive changed it since then

Reply:
Im happy with `HandleValueChanged`, it describes exactly when this method should be used, and it is private so there is no confusion.

Reply:
But it's not called when the Day value is changed, only Month and Year. If someone added for e.g. an OnValueChanged event call to this Handle function, they might not expect Day to be missed off.

Also it's only used to recalculate the days in the month, the general Handle functionality is still subscribed separately to each dropdown, even though it's shared logic. I think a `HandleValueChanged` function would be more suited as something to do this colour swap for all the dropdowns.